src/services/getUserService.js

```diff
@@ -0,0 +1,21 @@
+const { queryByUsername } = require("../repository/getUserDAO");
+
+
+async function getUserByUsername(req, res) {
+    const username = req.body.username;
+    const result = await queryByUsername(username);
+    const statusCode = result?.$metadata?.httpStatusCode;
+    const foundUser = result?.Item;
+    if (statusCode == 200) {
+        res.status(statusCode).json({
+            message: "User Found",
+            data: foundUser
+        });
+    } else {
+        res.status(statusCode).send();
```

Comment:
Restful API should use .json(). If it errors, put an empty bracket in the parenthesis. I would do something like res.status(statusCode).json({message: "error occurred for x reason"}) in this case, but you don't have to.

Reply:
I can default it to "non-unique ID generated", since that's the only case I've run into so far. Eventually we'd probably want some way to extract the error from the DynamoDB error message and build a message based on that.